os/board/rtl8721csm/src/component/os/tizenrt/rtk_netmgr.c

```diff
@@ -83,15 +83,81 @@ static wifi_utils_scan_list_s *g_scan_list;
 static int g_scan_num;
 extern struct netdev *ameba_nm_dev_wlan0;
 
+struct ap_scan_list {
+	unsigned int channel;
+	char ssid[WIFI_UTILS_SSID_LEN + 1];
+	unsigned int ssid_length;
+	int rssi;
+	wifi_utils_ap_auth_type_e ap_auth_type;
+	wifi_utils_ap_crypto_type_e ap_crypto_type;
+};
+typedef struct ap_scan_list ap_scan_list_s;
+
+static ap_scan_list_s *saved_scan_list;
+static uint32_t scan_number = 0;
+uint32_t ap_channel;
+
+#include "freertos/wrapper.h"
+#define SCAN_TIMER_DURATION 180000
+static _timer scan_timer;
+
+static void _scan_timer_handler(void *FunctionContext)
+{
+	ndbg("scan Timer expired : release saved scan list\r\n");
+	ndbg("scan Timer expired : sizeof(ap_scan_list_s) =%d, scan_number=%d \r\n", sizeof(ap_scan_list_s), scan_number);
+
+	rtw_mfree((unsigned char *)saved_scan_list, sizeof(ap_scan_list_s) * scan_number);
+	scan_number = 0;
+	rtw_del_timer(&(scan_timer));
+}
+
+static int save_scan_list(wifi_utils_scan_list_s *p_scan_list)
+{
+
+	// If application calls scan before scan result free(before timeout), release scan list and cancel timer
+	if (scan_number) {
+		rtw_mfree((unsigned char *)saved_scan_list, sizeof(ap_scan_list_s) * scan_number);
+		rtw_cancel_timer(&(scan_timer));
+		ndbg("scan is called before timeout\r\n");
+	} else {
+		rtw_init_timer(&(scan_timer), &(scan_timer), _scan_timer_handler, &(scan_timer), "dynamic_chk_timer");
+		ndbg("Start scan timer\n");
+	}
+
+	rtw_set_timer(&(scan_timer), SCAN_TIMER_DURATION);
+
+	scan_number	 = 0;
+	saved_scan_list = (ap_scan_list_s *)rtw_zmalloc(sizeof(ap_scan_list_s) * g_scan_num);

```

Comment:
Do you need to use zmalloc? After getting the pointer, you input values for entire members.
I think in your usecase, just malloc is better for the performance. Could you check?

Reply:
I'll change it to malloc and update with other review comments.

Reply:
@sunghan-chang @keerthanalex I think the ssid length varies, so that zmalloc might be better for cleaning used ssids or any garbages.

Reply:
Then I'll retain `rtw_zmalloc` in this case.

Reply:
@davidfather @keerthanalex There is the member, ssid_length so that it should be ok with non-initialized buffer.
```
        strncpy(saved_scan_list[scan_number].ssid, p_scan_list->ap_info.ssid, p_scan_list->ap_info.ssid_length);
        saved_scan_list[scan_number].ssid_length = p_scan_list->ap_info.ssid_length;
```
If we should use zmalloc, then I think `ssid_length` is unnecessary.

Reply:
@sunghan-chang, then we need to follow either one of the below
1) Retain zmalloc and remove `ssid_length` from struct `ap_scan_list`. The `ssid_length` parameter is not really used anywhere else apart from this place. 

- For writing to `ssid` of struct `ap_scan_list`, the `p_scan_list` ssid_length is used.
- In case of search in `wifi_netmgr_utils_connect_ap()`, the `ap_connect_config` ssid_length is used.

2) Modify  zmalloc to malloc and retain `ssid_length`. But as I mentioned in above point, the `ssid_length` is not used for any operation.
which one do you suggest?

Reply:
It's not proper to suggest ssid_length usages because I am not in charge of network. I just want to check necessity of zmalloc. But in this function point of view, I don't think zmalloc is necessary.

Reply:
@sunghan-chang, modified to malloc.

Reply:
@pillip8282 Could you confirm this?